### PR TITLE
store/copr: disable store batch copr for internal requests

### LINKED ISSUE #41562: Adding index on v6.6 is 5% slower than v6.5 when using tidb_ddl_enable_fast_reorg
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
add index on ossinsight dataset

### 2. What did you expect to see? (Required)
no performance regression

### 3. What did you see instead (Required)
5% performance regression 
### 4. What is your TiDB version? (Required)
v6.6

v6.6

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: close #41562

Problem Summary:

Add index has 5% regression in the latest tidb, and it's probably caused by batched copr.

### What is changed and how it works?

Disable batch for internal requests.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

I ran it 4 times with this PR and the time used are: 5453s, 5470s, 5459s, 5462s. Compare with the duration before, confirm that the regression is fixed.

image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: store/copr/coprocessor.go [hunk 1]
```diff
@@ -1953,3 +1944,23 @@ func optRowHint(req *kv.Request) bool {
...
+func checkStoreBatchCopr(req *kv.Request) bool {
+	if req.Tp != kv.ReqTypeDAG || req.StoreType != kv.TiKV {
+		return false
+	}
+	// TODO: support keep-order batch
+	if req.ReplicaRead != kv.ReplicaReadLeader || req.KeepOrder {
+		// disable batch copr for follower read
+		return false
+	}
+	// disable batch copr when paging is enabled.
```
Comment (Benjamin2037) [inline store/copr/coprocessor.go:1957]: Capitalization

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**